js_modules/dagit/packages/core/src/nav/RepositoryLocationStateObserver.tsx

```diff
@@ -63,6 +67,19 @@ export const RepositoryLocationStateObserver = () => {
     return null;
   }
 
+  if (flagLiveReload) {
+    setUpdatedLocations([]);
+    refetch();
+    client.resetStore();
+    SharedToaster.show({
+      icon: 'done',
+      intent: Intent.SUCCESS,
+      message: `Repository location ${updatedLocations[0]} has been updated and automatically reloaded.`,
+      timeout: 3000,
+    });
+  }
```

Comment:
This looks like behavior that shouldn't occur during render execution, but as a result of the subscription receiving data (or some other asynchronous behavior).

Regardless, even if it should occur independent of the subscription, we won't want to perform state updates (or reset the Apollo cache) in the render flow -- it would belong in a `useEffect` or something akin to that.